lib/Dialect/DC/Transforms/Analysis.cpp

```diff
@@ -0,0 +1,283 @@
+//===- Analysis.cpp - Analysis Pass -----------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Contains the definitions of the Analysis pass.
+//
+//===----------------------------------------------------------------------===//
+#include "circt/Dialect/DC/DCOps.h"
+#include "circt/Dialect/DC/DCPasses.h"
+#include "circt/Dialect/HW/HWDialect.h"
+#include "mlir/Pass/Pass.h"
+
+#include "circt/Dialect/Comb/CombDialect.h"
+#include "circt/Dialect/Comb/CombOps.h"
+#include "circt/Dialect/DC/DCOps.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Support/IndentedOstream.h"
+#include "mlir/Tools/mlir-translate/Translation.h"
+#include "llvm/ADT/ScopedHashTable.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/raw_ostream.h"
+#include <string>
+
+namespace circt {
+namespace dc {
+#define GEN_PASS_DEF_DCDOTPRINT
+#include "circt/Dialect/DC/DCPasses.h.inc"
+} // namespace dc
+} // namespace circt
+
+using namespace circt;
+using namespace dc;
+using namespace mlir;
+
+using ValueMap = llvm::ScopedHashTable<mlir::Value, std::string>;
+
+// stores operand and results for each node in the dot graph
+struct DotNode {
+  std::string nodeType;
+  llvm::SmallVector<std::pair<mlir::Value, std::string>> incoming;
+  llvm::SmallVector<std::pair<mlir::Value, std::string>> outgoing;
+};
+
+// gives unique name to each value in the graph
+llvm::SmallVector<std::pair<mlir::Value, std::string>>
+valueToName(const llvm::SmallVector<mlir::Value> &values,
+            llvm::SmallVector<std::pair<mlir::Value, std::string>> &currentMap,
+            bool tokenFlag) {
+  llvm::SmallVector<std::pair<mlir::Value, std::string>> res;
+  for (auto [i, v] : llvm::enumerate(values)) {
+    auto found = false;
+    for (const auto &cm : currentMap) {
+      if (v == cm.first) {
+        res.push_back({v, cm.second});
+        found = true;
+      }
+    }
+    if (!found) {
+      std::string name = "in_" + std::to_string(currentMap.size());
+      if (tokenFlag && i % 2 == 0)
+        name = "token_" + std::to_string(currentMap.size());
+      res.push_back({v, name});
+      currentMap.push_back({v, name});
+    }
+  }
+  return res;
+}
+
+DotNode createDCNode(
+    Operation &op,
+    llvm::SmallVector<std::pair<mlir::Value, std::string>> &valuesMap) {
+  if (auto castOp = llvm::dyn_cast<dc::ForkOp>(op))
```

Comment:
Nit: castOp never seems to be used so you can just do
```suggestion
  if (isa<dc::ForkOp>(op))
```